hashintel/hash
# H-373: Replace Next Ory API middleware with hash-api endpoint

## 🌟 What is the purpose of this PR?




This PR removes the Next.js API endpoint for proxying Ory kratos requests to the public Ory Kratos API endpoint, and replaces it with a proxy in the HASH API to eliminate lambda cold start as a potential cause of app slowness.

## 🔗 Related links





- H-373

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## 🛡 What tests cover this?



None

## ❓ How to test this?



1. Checkout the branch locally, and start the developer environment
2. Try to login, and check the user remains authenticated across refreshes. This tests both server-side Ory Kratos requests made by the FE (the `/whoami` request made to check if the user logged in), and browser side Ory Kratos requests (the `/login` request made to log the user in.

Unfortunately we don't have a way of testing this out in a staging environment, because it requires backend changes to be made and we don't have deployments for those in the staging environment. This is why the deployments attached to this PR will not work.

## 📹 Demo




https://github.com/hashintel/hash/assets/42802102/c1864cb5-dd6d-4199-a544-7d722a19d4d8
